### PR TITLE
space-cli registry, databse

### PR DESCRIPTION
**Pull request by KavishShah09**

### CODE REVIEW

File: space-cli/modules/addons/database.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	ctx := context.Background()
+
+	// Create a docker client
+	docker, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return utils.LogError("Unable to initialize docker client", "remove", "database", err)
+	}
+
+	// Check if a database container already exist
+	filterArgs := filters.Arg("label", "app=space-cloud")
```
Comment (YourTechBud) [inline space-cli/modules/addons/database.go]:
```suggestion
	filterArgs := filters.Arg("label", fmt.Sprintf("space-cloud--addon--db--%s", alias))
```

File: space-cli/modules/addons/database.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	containers, err := docker.ContainerList(ctx, types.ContainerListOptions{Filters: filters.NewArgs(filterArgs)})
+	if err != nil {
+		return utils.LogError("Unable to check if database already exists", "remove", "database", err)
+	}
+	if len(containers) == 0 {
+		utils.LogInfo("No space-cloud instance found. Run 'space-cli setup' first", "remove", "database")
```
Comment (YourTechBud) [inline space-cli/modules/addons/database.go]:
```suggestion
		utils.LogInfo(fmt.Sprintf("Database (%s) not found.", alias), "remove", "database")
```

File: space-cli/modules/addons/database.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+		// First start the container
+		if err := docker.ContainerStart(ctx, container.ID, types.ContainerStartOptions{}); err != nil {
+			return utils.LogError(fmt.Sprintf("Unable to start container (%s)", container.ID), "remove", "database", err)
+		}
+
+		// Get the container's info
+		info, err := docker.ContainerInspect(ctx, container.ID)
+		if err != nil {
+			return utils.LogError(fmt.Sprintf("Unable to inspect container (%s)", container.ID), "remove", "database", err)
+		}
```
Comment (YourTechBud) [inline space-cli/modules/addons/database.go]: Not required. Remove this

File: space-cli/modules/addons/database.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+		hostName := utils.GetServiceDomain(info.Config.Labels["service"], alias)
```
Comment (YourTechBud) [inline space-cli/modules/addons/database.go]:
```suggestion
		hostName := utils.GetServiceDomain("db", alias)
```

File: space-cli/modules/addons/registry.go [hunk 5]
```diff
@@ -56,9 +57,15 @@ func addRegistry(projectID string) error {
...
 
-		// TODO: Ask the user to select a projectID
-		projectID = projects[0].ID
-		utils.LogInfo(fmt.Sprintf("Adding registry to project - %s", projects[0].Name), "add", "registry")
+		var projectIDOptions []string
+		for _, projectInfo := range projects {
+			projectIDOptions = append(projectIDOptions, projectInfo.ID)
+		}
+
+		if err := survey.AskOne(&survey.Select{Message: "Select project ID", Options: projectIDOptions}, &projectID); err != nil {
```
Comment (YourTechBud) [inline space-cli/modules/addons/registry.go]: No need to ask if only one project was present

File: space-cli/modules/addons/actions.go [hunk 6]
```diff
@@ -95,6 +92,6 @@ func ActionAddDatabase(c *cli.Context) error {
 
 // ActionRemoveDatabase removes a database add on
 func ActionRemoveDatabase(c *cli.Context) error {
-	alias := c.GlobalString("alias")
+	alias := c.Args().Get(0)
```
Comment (YourTechBud) [inline space-cli/modules/addons/actions.go]: What if the user forgets to provide an argument

File: space-cli/modules/addons/registry.go [hunk 7]
```diff
@@ -57,15 +57,19 @@ func addRegistry(projectID string) error {
 			return utils.LogError("No projects found. Run this command after creating a project", err)
 		}
 
-		var projectIDOptions []string
-		for _, projectInfo := range projects {
-			projectIDOptions = append(projectIDOptions, projectInfo.ID)
+		projectID = projects[0].ID
+		if len(projects) > 1 {
```
Comment (YourTechBud) [inline space-cli/modules/addons/registry.go]: Check if length is zero? Throw an error requesting user to create a project first

File: space-cli/modules/addons/actions.go [hunk 8]
```diff
@@ -26,3 +53,45 @@ func ActionAddRegistry(c *cli.Context) error {
...
+
+// ActionRemoveRegistry removes a registry add on
+func ActionRemoveRegistry(c *cli.Context) error {
+	project := c.GlobalString("project")
+	return removeRegistry(project)
+}
+
+// ActionAddDatabase adds a database add on
+func ActionAddDatabase(c *cli.Context) error {
+	dbtype := c.Args().Get(0)
```
Comment (YourTechBud) [inline space-cli/modules/addons/actions.go]: What if user doesn't provide an argument. Throw an error in that case.

File: space-cli/modules/addons/registry.go (hunk 7 above)
Comment (KavishShah09) [reply space-cli/modules/addons/registry.go]: Already being done on line 56
Existing code

**Pull request merged**